site/app/libraries/database/DatabaseQueries.php

```diff
@@ -728,25 +728,65 @@ public function toggleLikes(int $post_id, string $current_user): array {
         try {
             $this->course_db->query("SELECT * FROM forum_upducks WHERE post_id = ? AND user_id = ?", [$post_id, $current_user]);
             $inDatabase = isset($this->course_db->rows()[0]);
+
+            $sqlFilteredCount = "SELECT COUNT(*) AS filtered_likes_count
+                             FROM forum_upducks f
+                             JOIN users u ON f.user_id = u.user_id
+                             WHERE f.post_id = ?
+                             AND u.user_group IN (1, 2, 3)";
             if ($inDatabase) {
                 $this->course_db->query("DELETE FROM forum_upducks WHERE post_id = ? AND user_id = ?", [$post_id, $current_user]);
+                $action = "unlike";
                 $this->course_db->query("SELECT COUNT(*) AS likes_count FROM forum_upducks WHERE post_id = ?", [$post_id]);
                 $likesCount = intval($this->course_db->rows()[0]['likes_count']);
-                return ['unlike',$likesCount];
+                $this->course_db->query($sqlFilteredCount, [$post_id]);
+                $filteredLikesCount = intval($this->course_db->rows()[0]['filtered_likes_count']);
             }
             else {
                 $this->course_db->query("INSERT INTO forum_upducks (post_id, user_id) VALUES (?, ?)", [$post_id, $current_user]);
+                $action = "like";
                 $this->course_db->query("SELECT COUNT(*) AS likes_count FROM forum_upducks WHERE post_id = ?", [$post_id]);
                 $likesCount = intval($this->course_db->rows()[0]['likes_count']);
-                return ['like',$likesCount];
+                $this->course_db->query($sqlFilteredCount, [$post_id]);
+                $filteredLikesCount = intval($this->course_db->rows()[0]['filtered_likes_count']);
```

Comment:
You can use `row()` to get a single row:
```suggestion
                $filteredLikesCount = intval($this->course_db->row()['filtered_likes_count']);
```